azurerm/internal/services/containers/kubernetes_cluster_pod_identity_resource_test.go

```diff
@@ -0,0 +1,239 @@
+
+func TestAccKubernetesClusterPodIdentity_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_cluster_pod_identity", "test")
+	r := KubernetesClusterPodIdentityResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (t KubernetesClusterPodIdentityResource) Exists(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	id, err := parse.ClusterID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Containers.KubernetesClustersClient.Get(ctx, id.ResourceGroup, id.ManagedClusterName)
+	if err != nil {
+		return nil, fmt.Errorf("reading Kubernetes Cluster (%s): %+v", id.String(), err)
+	}
+
+	if resp.ManagedClusterProperties == nil ||
+		resp.ManagedClusterProperties.PodIdentityProfile == nil ||
+		resp.ManagedClusterProperties.PodIdentityProfile.Enabled == nil ||
+		!*resp.ManagedClusterProperties.PodIdentityProfile.Enabled {
+		return utils.Bool(false), nil
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (r KubernetesClusterPodIdentityResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_kubernetes_cluster_pod_identity" "test" {
+  cluster_id = azurerm_kubernetes_cluster.test.id
+
+  pod_identity {
+    name        = "name1"
+    namespace   = "ns1"
+    identity_id = azurerm_user_assigned_identity.test.id
+  }
+
+  depends_on = [
+    azurerm_role_assignment.test
+  ]
+}
+`, r.template(data))
+}
+
+func (r KubernetesClusterPodIdentityResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_kubernetes_cluster_pod_identity" "import" {
+  cluster_id = azurerm_kubernetes_cluster_pod_identity.test.cluster_id
+
+  dynamic "pod_identity" {
+    for_each = azurerm_kubernetes_cluster_pod_identity.test.pod_identity
+    content {
+      name        = pod_identity.value.name
+      namespace   = pod_identity.value.namespace
+      identity_id = pod_identity.value.identity_id
+    }
+  }
+}
+`, r.basic(data))
+}
+
+func (r KubernetesClusterPodIdentityResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_kubernetes_cluster_pod_identity" "test" {
+  cluster_id = azurerm_kubernetes_cluster.test.id
+
+  pod_identity {
+    name        = "name1"
+    namespace   = "ns1"
+    identity_id = azurerm_user_assigned_identity.test.id
+  }
+
+  exception {
```

Comment:
Is it allow only expcetion but no identity? If so, can we add a new config that contains only the exception, and adding it as one step in the update test case?

Reply:
added